File: request.js
const http = require('http');
const fs = require('fs');
const path = require('path');

const fileName = 'README.md';
const boundary = '----MyCustomBoundary12345';
const filePath = path.join(__dirname, fileName);

const fileData = fs.readFileSync(filePath);

const bodyStr =
    `--${boundary}\r\n` +
    `Content-Disposition: form-data; name="file"; filename="${fileName}"\r\n` +
    `Content-Type: application/octet-stream\r\n\r\n` +
    fileData +
    `\r\n--${boundary}--\r\n`;

const options = {
    hostname: 'localhost',
    port: 9617,
    path: `/archer/file-api/file-upload?filename=${fileName}`,
    method: 'POST',
    headers: {
        'Content-Type': `multipart/form-data; boundary=${boundary}`,
        'Content-Length': bodyStr.length,
    },
};

const req = http.request(options, (res) => {
    res.on('data', (chunk) => {
        console.log(`response: ${chunk.toString()}`);
    });
});

req.write(bodyStr);
req.end();

// http://localhost:9617/archer/file-api/file-view?filename=README.md

File: request.py
from archernet import HttpStatusCode, HttpClient, HttpClientResponse, FormData
import json

# http://localhost:9617/archer/file-api/file-view?filename=README.md
def test_upload_file():
    form = FormData()
    form.put_file("file", "./README.md")
    
    res = HttpClient.post("http://127.0.0.1:9617/archer/file-api/file-upload?filename=README.md", 
        body=form)
    body = json.loads(str(res.content, encoding="UTF-8"))
    print(f"response: {body}")

test_upload_file()
